docs/data/material/components/snackbars/ToolpadNotifications.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+import {
+  useNotifications,
+  NotificationsProvider,
+} from '@toolpad/core/useNotifications';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Switch from '@mui/material/Switch';
+
+function NotifyRadioButton() {
+  const notifications = useNotifications();
+  const [online, setOnline] = React.useState(true);
+  const prevOnline = React.useRef(online);
+  React.useEffect(() => {
+    if (prevOnline.current === online) {
+      return () => {};
+    }
+    prevOnline.current = online;
+
+    // preview-start
+    const key = online
+      ? notifications.show('You are now online', {
+          severity: 'success',
+          autoHideDuration: 3000,
+        })
+      : notifications.show('You are now offline', {
+          severity: 'error',
+        });
+
+    return () => {
+      notifications.close(key);
+    };
+    // preview-end
+  }, [notifications, online]);
+
+  return (
+    <div>
+      <FormControlLabel
```

Comment:
Please no. This feels wrong. Notifications are pushed-based, not state-based. Can we have something where developers can spam-press the button to simulate a couple of notifications? 

```suggestion
```

going with a button should also help make the source of the demo simpler to follow.